gui-libs/display-manager-init/files/display-manager.initd-11

```diff
@@ -0,0 +1,235 @@
+			PIDFILE=/run/gpe-dm.pid
+			;;
+		lxdm)
+			EXE=/usr/sbin/lxdm-binary
+			PIDFILE=/run/lxdm.pid
+			START_STOP_ARGS="--background"
+			;;
+		lightdm)
+			EXE=/usr/sbin/lightdm
+			PIDFILE=/run/lightdm.pid
+			START_STOP_ARGS="--background"
+			;;
+		sddm)
+			EXE="/usr/bin/sddm"
+			START_STOP_ARGS="-m --background"
+			PIDFILE=/run/sddm.pid
+			;;
+		*)
+			# first find out if there is such executable
+			EXE="$(command -v ${MY_XDM} 2>/dev/null)"
+			PIDFILE="/run/${MY_XDM}.pid"
+
+			# warn user that they are doing sick things if the exe was not found
+			if [ -z "${EXE}" ]; then
+				echo "ERROR: Your XDM value is invalid."
+				echo "  No ${MY_XDM} executable could be found on your system."
+			fi
+			;;
+	esac
+
+	if ! [ -x "${EXE}" ]; then
+		EXE=/usr/bin/xdm
+		PIDFILE=/run/xdm.pid
+		if ! [ -x "/usr/bin/xdm" ]; then
+			echo "ERROR: Please set your DISPLAYMANAGER variable in /etc/conf.d/xdm,"
+			echo "	or install x11-apps/xdm package"
+			eend 255
+		fi
+	fi
+}
+
+# Check to see if something is defined on our VT
+vtstatic() {
+	if [ -e /etc/inittab ] ; then
+		grep -Eq "^[^#]+.*\<tty$1\>" /etc/inittab
+	elif [ -e /etc/ttys ] ; then
+		grep -q "^ttyv$(($1 - 1))" /etc/ttys
+	else
+		return 1
+	fi
+}
+
+start() {
+	local EXE NAME PIDFILE AUTOCLEAN_CGROUP
+	setup_dm
+
+	if [ -f /etc/.noxdm ]; then
+		einfo "Skipping ${EXE##*/}, /etc/.noxdm found or \"nox\" bootparam passed."
+		rm /etc/.noxdm
+		return 0
+	fi
+
+	ebegin "Setting up ${EXE##*/}"
+
+	# save the prefered DM
+	save_options "service" "${EXE}"
+	save_options "name"    "${NAME}"
+	save_options "pidfile" "${PIDFILE}"
+	save_options "start_stop_args" "${START_STOP_ARGS}"
+	save_options "autoclean_cgroup" "${AUTOCLEAN_CGROUP:-no}"
+
+	if [ -n "${CHECKVT-y}" ] ; then
+		if vtstatic "${CHECKVT:-7}" ; then
+			if [ -x /sbin/telinit ] && [ "${SOFTLEVEL}" != "BOOT" ] && [ "${RC_SOFTLEVEL}" != "BOOT" ]; then
+				ewarn "Something is already defined on VT ${CHECKVT:-7}, will start X later"
+				telinit a >/dev/null 2>&1
+				return 0
+			else
+				eerror "Something is already defined on VT ${CHECKVT:-7}, not starting"
+				return 1
+			fi
+		fi
+	fi
+
+	/usr/bin/startDM
+	eend 0
+}
+
+stop() {
+	local curvt retval
+
+	retval=0
+	if [ -t 0 ]; then
+		if type fgconsole >/dev/null 2>&1; then
+			curvt=$(fgconsole 2>/dev/null)
+		else
+			curvt=$(tty)
+			case "${curvt}" in
+				/dev/ttyv[0-9]*) curvt=${curvt#/dev/ttyv} ;;
+				*) curvt= ;;
+			esac
+		fi
+	fi
+	local myexe myname mypidfile myservice
+	myexe=$(get_options "service")
+	myname=$(get_options "name")
+	mypidfile=$(get_options "pidfile")
+	myservice=${myexe##*/}
+	yesno "${rc_cgroup_cleanup:-no}" || rc_cgroup_cleanup=$(get_options "autoclean_cgroup")
+
+	[ -z "${myexe}" ] && return 0
+
+	ebegin "Stopping ${myservice}"
+
+	if start-stop-daemon --quiet --test --stop --exec "${myexe}" 2>/dev/null; then
+		start-stop-daemon --stop --exec "${myexe}" --retry TERM/5/TERM/5 \
+			${mypidfile:+--pidfile} ${mypidfile} \
```

Comment:
It would be good to double quote those variables.

Reply:
done